Bring back rails_event_store_active_record for 2.x

lukaszreszke opened the pull request.

pawelpacana commented:
Good:

* resurrecting `gem "rails_event_store_active_record", path: "../../ruby_event_store-active_record"` to gemfiles, which tells bundler where to find local gem when such is needed (we do not load gems from rubygems in development)

* removing deprecation warning from `rails_event_store_active_record`

Bad:

* removing `gem "ruby_event_store-active_record", path: "../../ruby_event_store-active_record"` from gemfiles: that way bundler will be unable to resolve this dependency when needed, and we will need this since `rails_event_store_active_record` depends now on `ruby_event_store-active_record`

Ugly:

* not removing `rails_event_store_active_record` post-install message
* bringing back `rails` to gemfiles of `ruby_event_store-active_record` and those gemfiles to CI matrix


What was achieved in extracting `ruby_event_store-active_record` was that:
* we no longer needed Rails there, still providing conditional support
* conditional Rails support testing moved to `rails_event_store`

We went too far by setting unrealistic expectations of breaking change in Rails 2.x — and wanted to revert just that: no warns, no post-install messages, `rails_event_store` depending on `rails_event_store_active_record`. The rest of improvements stay.

In the near future, we'll be able to expect that `ruby_event_store-active_record` has a migration task and `rails_event_store` only wraps it for Rails convenience. But that is different task :)

lukaszreszke commented:
Thanks for the feedback. I introduced the changes that you pointed out.
- [Brought back dependency to `ruby_event_store-active_record` in Gemfiles](https://github.com/RailsEventStore/rails_event_store/pull/1481/commits/129b6c70285ed1e5fdab8a70be7eb8ef62edb384)
- Removed `rails` dependency [spec_helper](https://github.com/RailsEventStore/rails_event_store/pull/1481/commits/fce660b6467cdf2b551684c6d9eea640971923a8), [gemfile](https://github.com/RailsEventStore/rails_event_store/pull/1481/commits/046b5118765aa4ed9835f7de836e05d569c62cf2), [gemfile.lock](https://github.com/RailsEventStore/rails_event_store/pull/1481/commits/c6b0930ab3baaaa7fbd2c512d53717ba7b1ef98b)  and [dropped restored rails-related gemfiles](https://github.com/RailsEventStore/rails_event_store/pull/1481/commits/877188c172ee1ffb58eb7967f494692cf73e2500)

The post install message was removed [here](https://github.com/RailsEventStore/rails_event_store/commit/057e333e5d6fc3f82d9ba31dfbc8f4de846a8523). My bad. Should've been part of this PR or at least rebased.
